### ISSUE TITLE
Components abstract properties type question

### ISSUE THREAD

**Issue by couchemar** [ahrefs/bs-reactstrap#14]
We have Label component, which define some abstract types for properties:
https://github.com/ahrefs/bs-reactstrap/blob/master/src/BsReactstrap__Label.re#L5-L10

When I tried to pass this properties to component i got an error:

```
  18 ┆ <Form>
  19 ┆   <FormGroup row=true>
  20 ┆     <Label sm=2> {js|Label|js} </Label>
  21 ┆     <Col sm=10>
  22 ┆       <Input
  
  This has type:
    int
  But somewhere wanted:
    BsReactstrap.Label.sm (defined as BsReactstrap__Label.sm)
```

How can I pass this kind of properties to component?

**Comment by liubko** [comment]
@couchemar oops.. I think we broke things a bit when moved to the new record syntax

Right now you can:
- do `<Label sm=Obj.magic(2)>`
- rollback to previous version `0.2.2`, `<Label sm=2>` will work there


@rusty-key @sscaff1 how can we improve this? we don't want make consumer call `Obj.magic` all the time, right?

**Comment by couchemar** [comment]
Hi, @liubko thanks for quick answer!
I have already done rollback.
Thanks for `Obj.magic` never knew about it

**Comment by rusty-key** [comment]
I guess we should just define this types properly:
```
type sm;
↓
type sm = int;
```

**Comment by liubko** [comment]
@rusty-key 
1. doing properly is a bit harder https://github.com/reactstrap/reactstrap/blob/master/src/Label.js#L11
```
 PropTypes.oneOfType([
  PropTypes.string,
  PropTypes.number,
  PropTypes.shape({
    size: stringOrNumberProp,
    push: deprecated(stringOrNumberProp, 'Please use the prop "order"'),
    pull: deprecated(stringOrNumberProp, 'Please use the prop "order"'),
    order: stringOrNumberProp,
    offset: stringOrNumberProp,
  }),
]);
```

2. `sm` is not the only one

**Comment by liubko** [comment]
Reverted to the old syntax for now & published v0.4.0
